wasatch/FeatureIdentificationDevice.py

```diff
@@ -1304,243 +1441,184 @@ def set_tec_enable(self, flag):
+        return SpectrometerResponse(data=self.settings.state.selected_laser)
 
-    def get_laser_enabled(self):
-        flag = 0 != self.get_code(0xe2, label="GET_LASER_ENABLED", msb_len=1)
+    def get_laser_enabled(self) -> SpectrometerResponse:
+        flag = 0 != self._get_code(0xe2, label="GET_LASER_ENABLED", msb_len=1).data
         log.debug("get_laser_enabled: %s", flag)
         self.settings.state.laser_enabled = flag
-        return flag
-
-    ##
-    # Turn the laser on or off.
-    #
-    # If laser power hasn't yet been externally configured, applies the default
-    # of full-power.
-    #
-    # If the new laser state is on, AND if laser ramping has been enabled, then
-    # the function will internally use the (blocking) software laser ramping
-    # algorithm; otherwise the new state will be applied immediately.
-    #
-    # @param flag (Input) bool (True turns laser on, False turns laser off)
-    # @returns whether the new state was applied
-    def set_laser_enable(self, flag):
+        return SpectrometerResponse(data=flag)
+
+    def set_laser_enable(self, flag: bool) -> SpectrometerResponse:
+        """
+        Turn the laser on or off.
+        If laser power hasn't yet been externally configured, applies the default
+        of full-power.
+        If the new laser state is on, AND if laser ramping has been enabled, then
+        the function will internally use the (blocking) software laser ramping
+        algorithm; otherwise the new state will be applied immediately.
+        @param flag (Input) bool (True turns laser on, False turns laser off)
+        @returns whether the new state was applied
+        """
         if not self.settings.eeprom.has_laser:
             log.error("unable to control laser: EEPROM reports no laser installed")
-            return False
+            return SpectrometerResponse(data=False, error_msg="no laser installed")
 
         # ARM seems to require that laser power be set before the laser is enabled
         if self.next_applied_laser_power is None:
             self.set_laser_power_perc(100.0)
 
         self.settings.state.laser_enabled = flag
-        if flag and self.get_laser_power_ramping_enabled():
+        if flag and self.get_laser_power_ramping_enabled().data:
             self._set_laser_enable_ramp()
         else:
             self._set_laser_enable_immediate(flag)
-        return True
-
-    ##
-    # Enable software (blocking) laser power ramping algorithm.
-    # @param flag (Input) whether laser ramping is enabled (default False)
-    # @see _set_laser_enable_ramp
-    def set_laser_power_ramping_enable(self, flag):
+        return SpectrometerResponse(data=True)
+
+    def set_laser_power_ramping_enable(self, flag: bool) -> SpectrometerResponse:
+        """
+        Enable software (blocking) laser power ramping algorithm.
+        @param flag (Input) whether laser ramping is enabled (default False)
+        @see _set_laser_enable_ramp
```

Comment:
Feel free to remove the laser ramping code altogether (only use the _immediate functions).  This never worked well, and I don't think any current customers rely on it.